sssd_test_framework/utils/tools.py

```diff
@@ -515,6 +515,31 @@ def grep(self, pattern: str, paths: str | list[str], args: list[str] | None = No
 
         return command.rc == 0
 
+    def connected_interface(self) -> str:
+        """
+        Retrieves the currently connected network interface on a Linux system by executing the
+        'ip route show default' command on a remote host.
+
+        This method uses the `exec` method of the `host.conn` object to run the 'ip route show default'
+        command on the remote host. It then parses the output to determine the network interface
+        used for the default route.
+
+        Returns:
+            str: The name of the connected network interface (e.g., 'eth0', 'wlan0') if found.
+                 If the command fails, returns empty string.
+
+        Example:
+            >>> client.tools.connected_interface()
+            'eth0'
+        """
+        result = self.host.conn.exec(["ip", "route", "show", "default"])
+        output = jc.parse('ip_route', result.stdout)
+        if output and isinstance(output, list):
+            for route in output:
+                interface = route.get('dev')
+                if interface:
+                    return interface
```

Comment:
Hi,
we can expect the result to be a list, and we can assert that only one is returned since you  ask for the default one. The following shuold also fix PR CI.

```python
if not output or len(output) != 1:
    raise ValueError("Unexpected number of default interfaces")

interface = output[0].get('dev')
if not interface:
    raise ValueError("Default interface is not set")

return interface
```